sci-wcalc/ic_microstrip_syn.man

.\" Copyright (c), 2005 Dan McMahill
.\" Do not edit this directly.  Edit the XML source file instead
.\"

.TH ic_microstrip_syn "" "" "Wcalc" "Wcalc Commands"
.B ic_microstrip_syn
- Integrated circuit microstrip transmission line synthesis

.SH SYNOPSIS

[w_out, h_out, tox_out, l_out] = 
ic_microstrip_syn(z0, w, h, l, tmet, rho, rough, er, tand, f, flag)


.SH INPUT PARAMETERS

.TP 10
z0
: desired characteristic impedance (Ohms)
.TP 10
w
: width of ic_microstrip line (meters)
.TP 10
h
: substrate thickness (meters)
.TP 10
l
: lentgh of ic_microstrip line (meters)
.TP 10
tmet
: metal thickness (meters)
.TP 10
rho
: resistivity of wire (relative to copper)
.TP 10
rough
: conductor surface roughness (meters-rms)
.TP 10
er
: substrate relative permitivitty
.TP 10
tand
: substrate loss tangent
.TP 10
f
: synthesis frequency (Hz)
.TP 10
flag
: indicates which parameters should be optimized
    
.RS
.TP 10
flag=1
: synthesize the metal width
.TP 10
flag=1
: synthesize the substrate thickness
.TP 10
flag=2
: synthesize the oxide thickness
.RE


.SH OUTPUT VALUES

.TP 10
w_out
: calculated metal width
.TP 10
h_out
: calculated substrate thickness
.TP 10
tox_out
: calculated oxide thickness
.TP 10
l_out
: calculated metal length
.SH DESCRIPTION

Function to synthesize the physical dimenstions of a single
ic_microstrip transmission line to achieve a desired characteristic
impedance and electrical length.  The desired characteristic
impedance and some of the physical
dimensions of the ic_microstrip line are given as inputs and the
remaining dimensions are calculated.

The model accounts for dispersion (frequency dependent propagation
velocity).


.NF



                 |<--W-->|
                  _______    
                 | metal | <- tmet,rho,rough
    ----------------------------------------------
   {**************  Oxide (tox,eox) **************
    ----------------------------------------------
   (  substrate                          /|\     (
    )   es,sigmas                     H   |       )
   (                                     \|/     (
    ----------------------------------------------
    XXXXXXXXXXXXXXXXXX ground XXXXXXXXXXXXXXXXXXXX




.FI

.SH EXAMPLE
.nf

.TP
 // desired impedance
z0=75;
.TP
 // desired electrical length (degrees)
elen=90;
.TP
 // micron to meters conversion factor
sf=1.0e-6;
.TP
 // width 
w=160e-6;
.TP
 // length
l=1000e-6;
.TP
 // oxide thickness
tox=1.0e-6;
.TP
 // oxide relative dielectric constant
eox=4.0;
.TP
 // substrate thickness
h=250e-6;
.TP
 // substrate relative dielectric constant
es=11.8;
.TP
 // substrate conductivity (1/(ohm-m))
sigmas=10;
.TP
 // metal thickness
tmet=1.6e-6;
.TP
 // metal resitivity (ohm-m)
rho=3e-8;
.TP
 // metal surface roughness
rough=0;
.TP
 // frequency
f=2.4e9;
.TP
 // synthesize width
flag=0;
[w_out,h_out,tox_out,l_out] = ...
    ic_microstrip_syn(z0,elen,w,l,tox,eox,h,es,sigmas,tmet,rho,rough,f,flag);
disp(sprintf('w   = %8.3g um', w_out*1e6));
disp(sprintf('h   = %8.3g um', h_out*1e6));
disp(sprintf('tox = %8.3g um', tox_out*1e6));
disp(sprintf('l   = %8.3g um', l_out*1e6));
.fi
.SH SEE ALSO
stdio-wcalc(1),
air_coil_calc(), air_coil_syn(), coax_calc(), coax_syn(), coplanar_calc(), coplanar_syn(), coupled_microstrip_calc(), coupled_microstrip_syn(), coupled_stripline_calc(), coupled_stripline_syn(), ic_microstrip_calc(), ic_microstrip_syn(), microstrip_calc(), microstrip_syn(), stripline_calc(), stripline_syn()
.SH AUTHOR

Dan McMahill

.SH BUGS

None known


NEWS
$Id: NEWS,v 1.10 2009/02/28 03:26:18 dan Exp $

Wcalc -- history of user-visible changes. 

Copyright (c) 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005
2006, 2007, 2008, 2009 Dan McMahill

Please report bugs on the sourceforge project page.

===============
Version 1.1
===============

Added series/parallel RC equivalent circuit calculator.

Added series/parallel RL equivalent circuit calculator.

Added self and mutual inductance of two rectangular bars.

Added coplanar waveguide model.

Added coupled stripline model.

Corrected the Q calculations for low frequency (not skindepth) region
in air core inductors.  Also corrected Q calculation for conductors
other than copper.

Corrected the calculation of incremental conductance for the
microstrip model.

Corrected a bug in the calculation of conductor losses in the
microstrip model.

Added/enabled code for calculating conductor losses in the coupled
microstrip model.

When compiling with gtk-2.10 or newer, use gtkPrint for printing.
This gives a more professional/standard print dialog and also enables
printing under windows.

Added a preliminary Dutch translation.

Converted the figures to all use solid grayscale fills instead of
pattern fills as the former prints better.

Several updates to the build system to moderize it a bit.  Uses
AM_CONDITIONALS more appropriately, uses AC_PROG_LIBTOOL instead of
AM_PROG_LIBTOOL, etc.

Update the win32 build script to generate an installer that works
correctly under vista.

Add .wc file associations under vista.

Update the octave build infrastructure to work with octave-3.0 and
newer where mex support is built into mkoctfile instead of a
standalone mex shell script.

Switch to GPL licensing.

Create several utility functions to make it faster to build new gtk
forms.  This is only visible to developers.

Improved autogen.sh to check versions of several of the auto* tools to
help make it easier to diagnose any issues which may come up.

Repair compilation with the SunPRO compilers.


===============
Version 1.0
===============
Replaced the alert dialog box in the gtk version with a log
window.  This prevents a problem when many alerts are generated
and the user is presented with many many alert windows.

Added win32 support along with a build script for a windows installer.

Fixed a bug which gave lots of error/warning messages in the cgi
frontend for coupled microstrip.

===============
Version 0.9
===============
First public release.

===============
Version 0.1 
===============
Initial tagged version.  This does good analysis and synthesis
of microstrip.  Considered stable for this purpose.  However,
open/save/print and units other than mils are not supported yet.



octave-wcalc/ic_microstrip_syn.m
%IC_MICROSTRIP_SYN - Integrated circuit microstrip transmission line synthesis 
%  
%  SYNOPSIS
%  
%  [w_out, h_out, tox_out, l_out] = ic_microstrip_syn(z0, w, h, l, tmet, rho,
%  rough, er, tand, f, flag)
%  
%  PARAMETERS
%  
%  INPUT PARAMETERS
%  
%    * z0 : desired characteristic impedance (Ohms)
%    * w : width of ic_microstrip line (meters)
%    * h : substrate thickness (meters)
%    * l : lentgh of ic_microstrip line (meters)
%    * tmet : metal thickness (meters)
%    * rho : resistivity of wire (relative to copper)
%    * rough : conductor surface roughness (meters-rms)
%    * er : substrate relative permitivitty
%    * tand : substrate loss tangent
%    * f : synthesis frequency (Hz)
%    * flag : indicates which parameters should be optimized
%        + flag=1 : synthesize the metal width
%        + flag=1 : synthesize the substrate thickness
%        + flag=2 : synthesize the oxide thickness
%  
%  OUTPUT VALUES
%  
%    * w_out : calculated metal width
%    * h_out : calculated substrate thickness
%    * tox_out : calculated oxide thickness
%    * l_out : calculated metal length
%  
%  DESCRIPTION
%  
%  Function to synthesize the physical dimenstions of a single ic_microstrip
%  transmission line to achieve a desired characteristic impedance and electrical
%  length. The desired characteristic impedance and some of the physical
%  dimensions of the ic_microstrip line are given as inputs and the remaining
%  dimensions are calculated. The model accounts for dispersion (frequency
%  dependent propagation velocity).
%  
%  
%                   |<--W-->|
%                    _______
%                   | metal | <- tmet,rho,rough
%      ----------------------------------------------
%     {**************  Oxide (tox,eox) **************
%      ----------------------------------------------
%     (  substrate                          /|\     (
%      )   es,sigmas                     H   |       )
%     (                                     \|/     (
%      ----------------------------------------------
%      XXXXXXXXXXXXXXXXXX ground XXXXXXXXXXXXXXXXXXXX
%  
%  
%  
%  EXAMPLE
%  
%   % desired impedance
%  z0=75;
%   % desired electrical length (degrees)
%  elen=90;
%   % micron to meters conversion factor
%  sf=1.0e-6;
%   % width
%  w=160e-6;
%   % length
%  l=1000e-6;
%   % oxide thickness
%  tox=1.0e-6;
%   % oxide relative dielectric constant
%  eox=4.0;
%   % substrate thickness
%  h=250e-6;
%   % substrate relative dielectric constant
%  es=11.8;
%   % substrate conductivity (1/(ohm-m))
%  sigmas=10;
%   % metal thickness
%  tmet=1.6e-6;
%   % metal resitivity (ohm-m)
%  rho=3e-8;
%   % metal surface roughness
%  rough=0;
%   % frequency
%  f=2.4e9;
%   % synthesize width
%  flag=0;
%  [w_out,h_out,tox_out,l_out] = ...
%      ic_microstrip_syn(z0,elen,w,l,tox,eox,h,es,sigmas,tmet,rho,rough,f,flag);
%  disp(sprintf('w   = %8.3g um', w_out*1e6));
%  disp(sprintf('h   = %8.3g um', h_out*1e6));
%  disp(sprintf('tox = %8.3g um', tox_out*1e6));
%  disp(sprintf('l   = %8.3g um', l_out*1e6));
%  
%  SEE ALSO
%  
%  air_coil_calc
%  air_coil_syn
%  coax_calc
%  bars_calc
%  coax_syn
%  coplanar_calc
%  coplanar_syn
%  coupled_microstrip_calc
%  coupled_microstrip_syn
%  coupled_stripline_calc
%  coupled_stripline_syn
%  ic_microstrip_calc
%  ic_microstrip_syn
%  microstrip_calc
%  microstrip_syn
%  parallel_rc_p2s
%  parallel_rc_s2p
%  parallel_rl_p2s
%  parallel_rl_s2p
%  stripline_calc
%  stripline_syn
%  
%  AUTHOR
%  
%  Dan McMahill
%  
%  BUGS
%  
%  None known
%  
%  Please report any bugs on the Wcalc Sourceforge Project Page
%  


% Copyright (c) 2001-2009 Dan McMahill
% All rights reserved.
%
% This program is free software; you can redistribute it and/or modify
% it under the terms of the GNU General Public License as published by
% the Free Software Foundation; version 2 of the License.
% 
% This program is distributed in the hope that it will be useful,
% but WITHOUT ANY WARRANTY; without even the implied warranty of
% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
% GNU General Public License for more details.
% 
% You should have received a copy of the GNU General Public License
% along with this program; if not, write to the Free Software
% Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
%
